### PR TITLE
feat: jwt signature requests

### PR DESCRIPTION
**Pull request by thepiwo**
@CedrikNikita @davidyuk let me know what you think

### CODE REVIEW

File: src/popup/pages/SignMessage.vue [hunk 1]
```diff
@@ -39,7 +39,36 @@ export default defineComponent({
...
+              host,
+              name: host,
+              url: href,
+            },
+          });
+
+          const signature = await aeSdk.signMessage(message as string);
+          const signatureHex = Buffer.from(signature).toString('hex');
+          openCallbackOrGoHome(true, { signature: signatureHex });
+        }
```
Comment (CedrikNikita) [inline src/popup/pages/SignMessage.vue:71]:
```suggestion
        } else {
```
This is actually a copy of the code underneath.

File: package.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
   },
   "dependencies": {
     "@aeternity/aepp-calldata": "^1.5.1",
-    "@aeternity/aepp-sdk": "^13.2.2",
+    "@aeternity/aepp-sdk": "github:aeternity/aepp-sdk-js#feat/jwt-signature",
```
Comment (CedrikNikita) [inline package.json:35]: We would definitely need a proper release from the sdk, in order to include it in our release.

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 3]
```diff
@@ -92,6 +93,37 @@ export class AeAccountHdWallet extends AccountBase {
...
   }
 
+  override async signMessageJWT(
+    message: object,
+    options: Parameters<AccountBase['signMessageJWT']>[1] & InternalOptions,
+  ): Promise<string> {
+    if (options?.aeppOrigin) {
+      const { checkOrAskPermission } = usePermissions();
+      const permissionGranted = await checkOrAskPermission(
+        METHODS.signMessageJWT,
```
Comment (CedrikNikita) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:103]: Maybe we would need to come up with the new permission for this new method.

File: src/popup/pages/SignMessage.vue [hunk 4]
```diff
@@ -39,7 +39,36 @@ export default defineComponent({
         const message = isHexEncodedMessage ? Buffer.from(rawMessage, 'hex') : rawMessage;
         const displayMessage = message?.toString();
         const { host, href } = callbackOrigin.value || {} as any;
+        const jwt = route.query.jwt?.toString() === 'true';
 
+        if (jwt) {
+          const messageObject = JSON.parse(message as string);
+
+          await openModal(MODAL_MESSAGE_SIGN, {
```
Comment (CedrikNikita) [inline src/popup/pages/SignMessage.vue:47]: We would need either more info on existing modal or even new modal to be more user friendly for this action.

File: src/popup/pages/SignMessage.vue (hunk 1 above)
Comment (thepiwo) [reply src/popup/pages/SignMessage.vue:71]: yes it has been moved to the else case where the if is `if(jwt)`

File: src/popup/pages/SignMessage.vue (hunk 4 above)
Comment (thepiwo) [reply src/popup/pages/SignMessage.vue:47]: agree

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts (hunk 3 above)
Comment (thepiwo) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts:103]: agree

File: src/popup/pages/SignMessage.vue (hunk 1 above)
Comment (CedrikNikita) [reply src/popup/pages/SignMessage.vue:71]: It wasn't moved, yet. There are currently
```
if () {
  action a;
} else {
  action b;
}
action b;
```

Comment (thepiwo) [reply src/popup/pages/SignMessage.vue:71]: oh right, just noticed, thanks

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts (hunk 3 above)
Comment (thepiwo) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts:103]: isn't `METHODS.signMessageJWT` already its own permission?

Comment (CedrikNikita) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts:103]: No, the check will just return `false` every time we check, so the user would need to confirm every time.

File: src/popup/pages/SignMessage.vue (hunk 4 above)
Comment (thepiwo) [reply src/popup/pages/SignMessage.vue:47]: adjusted

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts (hunk 3 above)
Comment (thepiwo) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts:103]: can you then help me understand what is needed to be done?

Comment (CedrikNikita) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts:103]: As an MVP i would go with asking user every time. It's just something for the future improvements.

Comment (thepiwo) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts:103]: I mean technically, I'm not very familiar with the wallet implementation, could you help me with that.

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 5]
```diff
@@ -92,6 +93,36 @@ export class AeAccountHdWallet extends AccountBase {
     );
   }
 
+  override async signJWT(
+    message: object,
+    options: Parameters<AccountBase['signJWT']>[1] & InternalOptions,
+  ): Promise<`${string}.${string}`> {
```
Comment (peronczyk) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:99]: Maybe we could name the `${string}.${string}` type and save it somewhere in `types/index.tx` file so this code would be easier to read?

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 6]
```diff
@@ -92,6 +93,36 @@ export class AeAccountHdWallet extends AccountBase {
...
+      if (!permissionGranted) {
+        throw new RpcRejectedByUserError('Rejected by user');
+      }
+    }
+    const { getLastActiveProtocolAccount, getAccountByAddress } = useAccounts();
+    const account = (options?.fromAccount)
+      ? getAccountByAddress(options.fromAccount)
+      : getLastActiveProtocolAccount(PROTOCOLS.aeternity);
+
+    if (account && account.secretKey && account.protocol === PROTOCOLS.aeternity) {
```
Comment (peronczyk) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:116]:
```suggestion
    if (account?.secretKey && account?.protocol === PROTOCOLS.aeternity) {
```

**Comment by thepiwo** [comment]
replaced by https://github.com/superhero-com/superhero-wallet/pull/2952

**Pull request closed without merging**